src/main/java/org/wso2/lsp4intellij/client/languageserver/requestmanager/DefaultRequestManager.java

```diff
@@ -427,7 +427,7 @@ public CompletableFuture<Hover> hover(HoverParams params) {
             try {
                 return
                         Optional.ofNullable(serverCapabilities.getHoverProvider())
-                                .map(e -> e.getLeft() || e.getRight() != null).orElse(false) ?
+                                .map(e -> e.getRight() != null || e.getLeft()).orElse(false) ?
```

Comment:
```suggestion
                                .map(e -> e.getRight() != null || (e.getLeft() != null && e.getLeft())).orElse(false) ?
```

Reply:
@jczuchnowski since `e.getLeft()` is a a `Boolean` object, we still might have to consider the possibility of it being null IMO. So shall we do the checks in the above way?

Reply:
If both left and right is null then that would be a violation of the protocol. So I suppose there are two solutions to consider:
1. We assume false (as per your suggestion)
2. We throw a meaningful exception about the broken protocol.

I'm just wondering if 1. isn't misleading somehow. Eg. developer makes a mistake - wants to return true, but returns null. There's no error, everything works, but not as it's supposed to. Not sure if that's a valid concern, though. I you think defaulting to false is an acceptable solution, then I'm for it.

Reply:
Ok, let's go with false by default.